src/librustc_typeck/outlives/utils.rs

```diff
@@ -161,9 +162,21 @@ fn is_free_region(tcx: TyCtxt<'_>, region: Region<'_>) -> bool {
         // ignore it.  We can't put it on the struct header anyway.
         RegionKind::ReLateBound(..) => false,
 
+        // This can appear with malformed code (#64855):
+        //
+        //     struct Bar<T>(<Self as Foo>::Type) where Self: ;
+        //
+        // We accept it only to avoid an ICE.
+        RegionKind::ReEmpty => {
+            tcx.sess.delay_span_bug(
+                DUMMY_SP,
+                &format!("unexpected region in outlives inference: {:?}", region),
+            );
```

Comment:
This can also happen in well-formed code: `pub struct Bar<'a>(&'a Self) where Self: ;`.
Can you add a check-pass test for this as well and adjust the comment?